### PR TITLE
Remove VSCode extension host

### PR DESCRIPTION
**Pull request by smolck**

### CODE REVIEW

File: src/core/instance-api.ts [hunk 1]
```diff
@@ -208,32 +208,32 @@ const nvimRemoveHighlightSearch = async (
...
-    new Proxy(Object.create(null), {
-      get: (_: any, method: string) => (...args: any[]) => {
-        const res = Reflect.get(manualAI, namespace)
-        const manualFn = res && Reflect.get(res, method)
-        if (manualFn) return manualFn(...args)
-        ee.on(`ai.${namespace}.${method}`, args[0])
-      },
-    }),
-})
+// const manualAI = {
```
Comment (smolck) [inline src/core/instance-api.ts]: TODO: Remove comment

File: src/core/instance-api.ts [hunk 2]
```diff
@@ -6,12 +6,12 @@ import {
...
 import { VimMode, BufferInfo, HyperspaceCoordinates } from '../neovim/types'
 import { MessageStatusUpdate, MessageReturn } from '../protocols/veonim'
-import { AIClient, WorkspaceSymbol } from '../ai/protocol'
+// import { AIClient, WorkspaceSymbol } from '../ai/protocol'
 import { onFnCall, pascalCase } from '../support/utils'
 import { colors } from '../render/highlight-attributes'
 import { Functions } from '../neovim/function-types'
 import { WindowMetadata } from '../windows/metadata'
-import { CompletionItem } from '../components/autocomplete'
+// import { CompletionItem } from '../components/autocomplete'
```
Comment (smolck) [inline src/core/instance-api.ts]: Remove comment

File: src/core/instance-api.ts [hunk 3]
```diff
@@ -6,12 +6,12 @@ import {
 } from '../core/instance-manager'
 import { VimMode, BufferInfo, HyperspaceCoordinates } from '../neovim/types'
 import { MessageStatusUpdate, MessageReturn } from '../protocols/veonim'
-import { AIClient, WorkspaceSymbol } from '../ai/protocol'
+// import { AIClient, WorkspaceSymbol } from '../ai/protocol'
```
Comment (smolck) [inline src/core/instance-api.ts]: remove comment

File: src/components/references.ts [hunk 4]
```diff
@@ -264,10 +265,10 @@ const view = ($: S, a: typeof actions) =>
...
 
-const ui = app({ name: 'references', state, actions, view })
+export const ui = app({ name: 'references', state, actions, view })
 
-api.ai.references.onShow(
-  (references: ReferenceResult[], referencedSymbol?: string) => {
-    ui.show({ references, referencedSymbol })
-  }
-)
+// api.ai.references.onShow(
```
Comment (smolck) [inline src/components/references.ts]: remove comment (just here as a reminder of what was going on before)

File: src/components/autocomplete.ts [hunk 5]
```diff
@@ -227,14 +226,14 @@ const actions = {
...
 
-    if (!detail || !documentation)
-      (async () => {
-        // TODO: what do with .detail?
-        const details = await api.ai.completions.getDetail(completionItem)
-        if (!details || !details.documentation) return
-        const richFormatDocs = await parseDocs(details.documentation)
-        a.showDocs(richFormatDocs)
-      })()
+    // if (!detail || !documentation)
```
Comment (smolck) [inline src/components/autocomplete.ts:221]: remove comment (if it's unnecessary to keep the code; I just didn't know/don't know what it was doing)

File: src/components/autocomplete.ts [hunk 6]
```diff
@@ -1,21 +1,116 @@
...
 import Overlay from '../components/overlay'
 import * as Icon from 'hyperapp-feather'
-import { MarkdownString } from 'vscode'
 import { cursor } from '../core/cursor'
-import api from '../core/instance-api'
 import { h, app } from '../ui/uikit'
 
+export enum CompletionSource {
+  Neovim,
+  VSCode,
```
Comment (clason) [inline src/components/autocomplete.ts]: Is this still needed?

File: src/components/autocomplete.ts [hunk 7]
```diff
@@ -69,10 +164,10 @@ const getCompletionIcon = (kind: CompletionItemKind) =>
 
 // TODO: move to common place. used in other places like signature-hint
 const parseDocs = async (
-  docs?: string | MarkdownString
```
Comment (clason) [inline src/components/autocomplete.ts:72]: signature hints can contain Markdown, can't they?

File: src/components/autocomplete.ts (hunk 6 above)
Comment (smolck) [reply src/components/autocomplete.ts]: Ah, good point, probably not.

File: src/components/autocomplete.ts (hunk 5 above)
Comment (clason) [reply src/components/autocomplete.ts:221]: I think getting documentation may be slow (you don't want to get it for _all_ of the possibly many returned results at the same time), so async does make sense.

File: src/components/autocomplete.ts (hunk 7 above)
Comment (smolck) [reply src/components/autocomplete.ts:72]: Yes, but I think the markdown is just going to be in regular strings. If I understand correctly, `MarkdownString` is a type that is only provided from VSCode completions . . . although I could be wrong.

**Pull request merged**